### PR TITLE
Create login and register page #1

### PR DESCRIPTION
**Pull request by GabiGrad05**
**Developed by:**

- created the login and register module
- added forms and created validation using Reactive Forms
- customized the login and register page using custom classes
- created the routing for the login module

login

register

register-val

**WIP:**
- create the route to the register module
- change the login-register module name into authentication

### CODE REVIEW

File: apps/login-register/src/app/login/login.component.html [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          <div class="email">
+            <label>Email or UserName</label>
+            <input
+              class="email-input"
+              type="text"
+              formControlName="email"
+              placeholder="Email"
+            />
+            <span
+              *ngIf="(loginForm.controls['email'].touched || loginForm.controls['email'].dirty) && loginForm.controls['email'].errors?.['required']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/login/login.component.html]: You can move this to the component.ts or into a shared service to improve readability and code

File: apps/login-register/src/app/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+              class="err-message"
+            >
+              Email required
+            </span>
+            <span
+              *ngIf="(loginForm.controls['email'].touched || loginForm.controls['email'].dirty) && loginForm.controls['email'].errors?.['email']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/login/login.component.html]: here too

File: apps/login-register/src/app/login/login.component.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          <div class="password">
+            <label>Password</label>
+            <input
+              class="password-input"
+              type="password"
+              formControlName="password"
+              placeholder="Password"
+            />
+            <span
+              *ngIf="(loginForm.controls['password'].touched || loginForm.controls['password'].dirty) && loginForm.controls['password'].errors?.['required']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/login/login.component.html]: same

File: apps/login-register/src/app/login/login.component.html [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+              class="err-message"
+            >
+              This field is required
+            </span>
+
+            <span
+              *ngIf="(loginForm.controls['password'].touched || loginForm.controls['password'].dirty) && loginForm.controls['password'].errors?.['minlength']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/login/login.component.html]: same here

File: apps/login-register/src/app/register/register.component.html [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            <div class="first-name">
+              <label>First Name</label>
+              <input
+                class="firstName-input"
+                type="text"
+                formControlName="firstName"
+                placeholder="First Name"
+              />
+              <span
+                *ngIf="(registerForm.controls['firstName'].touched || registerForm.controls['firstName'].dirty) && registerForm.controls['firstName'].errors?.['required']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/register/register.component.html]: same as on the login component

File: apps/login-register/src/app/register/register.component.html [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            <div class="last-name">
+              <label>Last Name</label>
+              <input
+                class="lastName-input"
+                type="text"
+                formControlName="lastName"
+                placeholder="Last Name"
+              />
+              <span
+                *ngIf="(registerForm.controls['lastName'].touched || registerForm.controls['lastName'].dirty) && registerForm.controls['lastName'].errors?.['required']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/register/register.component.html]: here as well

File: apps/login-register/src/app/register/register.component.html [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            <div class="email">
+              <label>Email</label>
+              <input
+                class="email-input"
+                type="text"
+                formControlName="email"
+                placeholder="Email"
+              />
+              <span
+                *ngIf="(registerForm.controls['email'].touched || registerForm.controls['email'].dirty) && registerForm.controls['email'].errors?.['required']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/register/register.component.html]: same

File: apps/login-register/src/app/register/register.component.html [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+                class="err-message"
+              >
+                Email required
+              </span>
+              <span
+                *ngIf="(registerForm.controls['email'].touched || registerForm.controls['email'].dirty) && registerForm.controls['email'].errors?.['email']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/register/register.component.html]: here

File: apps/login-register/src/app/register/register.component.html [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            <div class="password">
+              <label>Password</label>
+              <input
+                class="password-input"
+                type="password"
+                formControlName="password"
+                placeholder="Password"
+              />
+              <span
+                *ngIf="(registerForm.controls['password'].touched || registerForm.controls['password'].dirty) && registerForm.controls['password'].errors?.['required']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/register/register.component.html]: here

File: apps/login-register/src/app/register/register.component.html [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+                class="err-message"
+              >
+                This field is required
+              </span>
+  
+              <span
+                *ngIf="(registerForm.controls['password'].touched || registerForm.controls['password'].dirty) && registerForm.controls['password'].errors?.['minlength']"
```
Comment (stanasaaccess) [inline apps/login-register/src/app/register/register.component.html]: and here

File: apps/login-register/src/app/types/messages.model.ts [hunk 11]
```diff
@@ -0,0 +1,6 @@
+export interface IMessages {
+    firstNameMessages?: string,
+    lastNameMessages?: string
+    emailMessages: string,
+    passwordMessages: string
+}
```
Comment (alexPopNicolae123) [inline apps/login-register/src/app/types/messages.model.ts]: add end line at the end of the file

File: apps/login-register/src/app/types/validation.model.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    emailMessages: {
+        required: string,
+        email: string
+    },
+    passwordMessages: {
+        required: string,
+        minlength: string
+    }
+}
```
Comment (alexPopNicolae123) [inline apps/login-register/src/app/types/validation.model.ts]: add end line at the end of the file

File: apps/login-register/src/app/register/register.component.ts [hunk 13]
```diff
@@ -9,6 +11,30 @@ import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 export class RegisterComponent implements OnInit {
   registerForm: FormGroup = new FormGroup({});
 
+  errorMessages: any = {
```
Comment (alexPopNicolae123) [inline apps/login-register/src/app/register/register.component.ts]: extract this into a constats.ts file to clear the cluttering

File: apps/login-register/src/app/login/login.component.ts [hunk 14]
```diff
@@ -10,6 +12,22 @@ import { Router } from '@angular/router';
 export class LoginComponent implements OnInit {
   loginForm: FormGroup = new FormGroup({});
 
+  errorMessages: any = {
+    emailMessages: '',
```
Comment (alexPopNicolae123) [inline apps/login-register/src/app/login/login.component.ts]: extract this into a separate file constants.ts

File: apps/login-register/src/app/login/login.component.ts [hunk 15]
```diff
@@ -41,5 +59,26 @@ export class LoginComponent implements OnInit {
       email: ['', [Validators.required, Validators.email]],
       password: ['', [Validators.required, Validators.minLength(8)]],
     });
+
```
Comment (alexPopNicolae123) [inline apps/login-register/src/app/login/login.component.ts]: extract this into a separate method and use it in the onInit

File: apps/login-register/src/app/register/register.component.ts [hunk 16]
```diff
@@ -34,5 +60,34 @@ export class RegisterComponent implements OnInit {
       email: ['', [Validators.required, Validators.email]],
       password: ['', [Validators.required, Validators.minLength(8)]],
     });
+
+    const firstName = this.registerForm.get('firstName');
```
Comment (alexPopNicolae123) [inline apps/login-register/src/app/register/register.component.ts]: extract this into a separate method and use it in the onInit

File: apps/login-register/src/app/login/login.component.html (hunk 1 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/login/login.component.html]: done

File: apps/login-register/src/app/login/login.component.html (hunk 2 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/login/login.component.html]: done

File: apps/login-register/src/app/login/login.component.html (hunk 3 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/login/login.component.html]: done

File: apps/login-register/src/app/login/login.component.html (hunk 4 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/login/login.component.html]: done

File: apps/login-register/src/app/register/register.component.html (hunk 5 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.html]: done

File: apps/login-register/src/app/register/register.component.html (hunk 6 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.html]: done

File: apps/login-register/src/app/register/register.component.html (hunk 7 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.html]: done

File: apps/login-register/src/app/register/register.component.html (hunk 8 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.html]: done

File: apps/login-register/src/app/register/register.component.html (hunk 9 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.html]: done

File: apps/login-register/src/app/register/register.component.html (hunk 10 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.html]: done

File: apps/login-register/src/app/types/messages.model.ts (hunk 11 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/types/messages.model.ts]: I changed the logic there

File: apps/login-register/src/app/register/register.component.ts (hunk 13 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.ts]: I tried but doesn't work like expected, so I changed the logic there

File: apps/login-register/src/app/login/login.component.ts (hunk 14 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/login/login.component.ts]: same

File: apps/login-register/src/app/login/login.component.ts (hunk 15 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/login/login.component.ts]: it is already in a private method which is called in the onInit cycle

image

image

File: apps/login-register/src/app/register/register.component.ts (hunk 16 above)
Comment (GabiGrad05) [reply apps/login-register/src/app/register/register.component.ts]: same

**Pull request merged**